src/main/java/de/dennisguse/opentracks/publicapi/StartRecording.java

```diff
@@ -1,11 +1,25 @@
 package de.dennisguse.opentracks.publicapi;
 
+import android.os.Bundle;
+
+import de.dennisguse.opentracks.data.models.Track;
 import de.dennisguse.opentracks.services.TrackRecordingService;
+import de.dennisguse.opentracks.settings.PreferencesUtils;
+import de.dennisguse.opentracks.util.IntentDashboardUtils;
 
 public class StartRecording extends AbstractAPIActivity {
+    private static final String TAG = StartRecording.class.getSimpleName();
 
     protected void execute(TrackRecordingService service) {
-        service.startNewTrack();
+        Track.Id trackId = service.startNewTrack();
```

Comment:
One thing: what happens if the trackId is null?
This might happen if already a recording is running and then we should probably not send the data, right?

Reply:
Agreed. Added an extra check for that.